tests/forms_tests/tests/test_forms.py

```diff
@@ -2486,6 +2486,14 @@ class FileForm(forms.Form):
         self.assertEqual(f.errors, {})
         self.assertEqual(f.cleaned_data['file1'], 'resume.txt')
 
+    def test_fileinput_use_required_attribute(self):
+        """#31118 - remove `required` attribute if initial data exists"""
+        class FileForm(Form):
+            file1 = forms.FileField(initial=lambda: 'resume.txt', widget=FileInput)
```

Comment:
Please move initial to the form instance (we're not testing a callable `initial` here, so this can be misleading), .e.g
```python
class FileForm(Form):
    file1 = FileField()

f = FileForm(initial={'file1': 'resume.txt'}, auto_id=False)
self.assertHTMLEqual(....
```
I added `auto_id=False` to remove unrelated HTML attribute.